src/AST/Translator.Expressions.dfy

```diff
@@ -451,6 +455,8 @@ module Bootstrap.AST.Translator {
   {
     if c is C.IdentifierExpr then
       TranslateIdentifierExpr(c as C.IdentifierExpr)
+    else if c is C.ConversionExpr then
+      TranslateUnsupportedExpression(c)
```

Comment:
Because it's a subtype of the following type that we don't support, so we have to check for it before the latter.

Reply:
Based on our conversation today we decided to change the test for UnaryExpr into one for UnaryOpExpr and also make sure that we never have a `Need` on the type, so that we only reject malformed expressions rather than unknown ones.

Reply:
Let's make an issue for this

Reply:
#15